src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -136,6 +136,15 @@ public override void OnStartActivity(Activity activity, object payload)
                     activity.SetTag(SemanticConventions.AttributeHttpUserAgent, userAgent);
                 }
 
+                if (this.options.RecordHttpServerAttributes)
+                {
+                    var xForwardedFor = request.Headers["X-Forwarded-For"].FirstOrDefault();
+                    if (!string.IsNullOrEmpty(xForwardedFor))
+                    {
+                        activity.SetTag(SemanticConventions.AttributeHttpClientIP, xForwardedFor.Split(',').First().Trim());
+                    }
```

Comment:
Should we add an `else` block that sends `HttpContext.Connection.RemoteIpAddress` as ClientIP if `X-Forwarded-For` isn't available?

Reply:
Added

Reply:
@lupengamzn Have you tested this change with AspNetCore [ForwardedHeadersMiddleware](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.httpoverrides.forwardedheadersmiddleware?view=aspnetcore-5.0)? It seems ti remove the X-Forwarded-For header after it is processed: https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs#L324